google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/InstanceAdminRequests.java

```diff
@@ -0,0 +1,585 @@
+      this.id = instanceId;
+      protoInstance.setDisplayName(displayName).setType(instanceType);
+    }
+
+    private Instance(com.google.bigtable.admin.v2.Instance instance) {
+      this.protoInstance = instance.toBuilder();
+      id = getName().getInstance();
+    }
+
+    public String getid() {
+      return id;
+    }
+
+    public InstanceName getName() {
+      return InstanceName.parse(protoInstance.getName());
+    }
+
+    public String getDisplayName() {
+      return protoInstance.getDisplayName();
+    }
+
+    public State getState() {
+      return protoInstance.getState();
+    }
+
+    public Type getType() {
+      return protoInstance.getType();
+    }
+
+    public Map<String, String> getLabelsMap() {
+      return protoInstance.getLabelsMap();
+    }
+
+    public Instance addLabel(String key, String value) {
+      // TODO: add an regex check. Error from is hard to decipher
+      protoInstance.putLabels(key, value);
+      return this;
+    }
+
+    @InternalApi
+    public com.google.bigtable.admin.v2.Instance toProto() {
+      return protoInstance.build();
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("name", getName())
+          .add("displayName", getDisplayName())
+          .add("state", getState())
+          .add("type", getType())
+          .add("labels", getLabelsMap())
+          .toString();
+    }
+  }
+
+  public static final class UpdateInstance {
+    private final String DISPLAY_NAME_FIELDMASK = "display_name";
+    private final String TYPE_FIELDMASK = "type";
+    private final String LABELS_FIELDMASK = "labels";
+
+    private final List<String> replacefields = new ArrayList<>();
+    private com.google.bigtable.admin.v2.Instance.Builder protoUpdateInstance =
+        com.google.bigtable.admin.v2.Instance.newBuilder();
+
+    private UpdateInstance(Instance instance) {
+      protoUpdateInstance = instance.toProto().toBuilder().clone();
+    }
+
+    public UpdateInstance updateDisplayName(String updatedDisplayName) {
+      replacefields.add(DISPLAY_NAME_FIELDMASK);
+      protoUpdateInstance.setDisplayName(updatedDisplayName);
+      return this;
+    }
+
+    public UpdateInstance upgradeType() {
+      Preconditions.checkArgument(
+          Type.DEVELOPMENT.equals(protoUpdateInstance.getType()),
+          "PRODUCTION type cannot be upgraded");
+      replacefields.add(TYPE_FIELDMASK);
+      protoUpdateInstance.setType(Type.PRODUCTION);
+      return this;
+    }
+
+    public UpdateInstance updateLabels(Map<String, String> updatedLabels) {
+      replacefields.add(LABELS_FIELDMASK);
+      protoUpdateInstance.putAllLabels(updatedLabels);
+      return this;
+    }
+
+    @InternalApi
+    public com.google.bigtable.admin.v2.Instance toProto() {
+      return protoUpdateInstance.clearState().build();
+    }
+
+    @InternalApi
+    public FieldMask getPartialUpdateFieldMask() {
```

Comment:
Might be better to have toProto return the UpdateInstanceRequest proto and merge these 2 methods